// @ant-design/react-native/lib/locale-provider/zh_CN.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _zh_CN = require('../date-picker-view/locale/zh_CN');

var _zh_CN2 = _interopRequireDefault(_zh_CN);

var _zh_CN3 = require('../date-picker/locale/zh_CN');

var _zh_CN4 = _interopRequireDefault(_zh_CN3);

var _zh_CN5 = require('../input-item/locale/zh_CN');

var _zh_CN6 = _interopRequireDefault(_zh_CN5);

var _zh_CN7 = require('../list-view/locale/zh_CN');

var _zh_CN8 = _interopRequireDefault(_zh_CN7);

var _zh_CN9 = require('../modal/locale/zh_CN');

var _zh_CN10 = _interopRequireDefault(_zh_CN9);

var _zh_CN11 = require('../pagination/locale/zh_CN');

var _zh_CN12 = _interopRequireDefault(_zh_CN11);

var _zh_CN13 = require('../picker/locale/zh_CN');

var _zh_CN14 = _interopRequireDefault(_zh_CN13);

var _zh_CN15 = require('../search-bar/locale/zh_CN');

var _zh_CN16 = _interopRequireDefault(_zh_CN15);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

exports['default'] = {
    locale: 'zh_CN',
    DatePicker: _zh_CN4['default'],
    DatePickerView: _zh_CN2['default'],
    InputItem: _zh_CN6['default'],
    Modal: _zh_CN10['default'],
    Pagination: _zh_CN12['default'],
    Picker: _zh_CN14['default'],
    SearchBar: _zh_CN16['default'],
    ListView: _zh_CN8['default']
};
module.exports = exports['default'];
